typescript/sdk/src/gas/HyperlaneIgpDeployer.ts

```diff
@@ -113,3 +207,64 @@ export class HyperlaneIgpDeployer extends HyperlaneDeployer<
     };
   }
 }
+
+// recursively fetches storage gas oracle configs from core config
+// eg. test1: core.defaultHook.igpConfig.test1.oracleConfig
+export function getStorageGasOracleConfigs(
+  coreConfig: ChainMap<CoreConfig>,
+): ChainMap<StorageGasOraclesConfig> {
+  const storageGasOracleConfigs: ChainMap<StorageGasOraclesConfig> = {};
+  for (const chain of Object.keys(coreConfig)) {
+    storageGasOracleConfigs[chain] = getStorageGasOracleConfig(
+      coreConfig[chain],
+    );
+  }
+  return storageGasOracleConfigs;
+}
+
+function getStorageGasOracleConfig(
+  coreConfig: CoreConfig,
+): StorageGasOraclesConfig {
+  const defaultIgpConfigs = getNestedIgpConfigs(coreConfig.defaultHook);
+  const requiredIgpConfigs = getNestedIgpConfigs(coreConfig.requiredHook);
+
+  const totalIgpConfigs = defaultIgpConfigs.concat(requiredIgpConfigs);
+  if (totalIgpConfigs.length === 0 || totalIgpConfigs.length > 1) {
+    throw Error(
+      `Incorrect number (${totalIgpConfigs.length}) of IGP configs found in core config. Please check your config.`,
+    );
+  } else if (totalIgpConfigs.length == 1 && requiredIgpConfigs.length > 0) {
```

Comment:
Recommend triple-equals comparison whenever possible. I thought the linter caught this but maybe not

Reply:
fixed